src/main/java/com/autotune/analyzer/deployment/AutotuneDeployment.java

```diff
@@ -545,7 +559,11 @@ private static AutotuneConfig getAutotuneConfig(String autotuneConfigResource, K
 
 				Tunable tunable;
 				try {
-					tunable = new Tunable(tunableName, step, upperBound, lowerBound, tunableValueType, queriesMap, sloClassList, layerName);
+					// check if categoricalValueList is empty and then only pass the upper, lower bounds else skip it
+					if(categoricalValueList.isEmpty())
```

Comment:
This needs to check for `categoricalValueType`. In the constructor it should fail if the list is empty or null